File: ios/Classes/EasytbkPlugin.m
#import "EasytbkPlugin.h"

@implementation EasytbkPlugin

EasytbkLoginService *_loginService;
EasytbkTradeHandler *_tradeHandler;


+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"easytbk"
            binaryMessenger:[registrar messenger]];
  EasytbkPlugin* instance = [[EasytbkPlugin alloc] init];
//   [registrar addMethodCallDelegate:instance channel:channel];
  [registrar addApplicationDelegate:instance];

}

// 初始化
- (instancetype)initWithRegistrar:(NSObject <FlutterPluginRegistrar> *)registrar {
    self = [super init];

    if (self) {
        _loginService = [[EasytbkLoginService alloc] init];
        _tradeHandler = [[EasytbkTradeHandler alloc] init];
    }

    return self;
}



- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {

     if ([@"initTradeAsync" isEqualToString:call.method]) {
        [_tradeHandler initTradeAsync:call result:result];
    } else if ([@"login" isEqualToString:call.method]) {
        [_loginService handleLogin:call result:result];
    } else if ([@"isLogin" isEqualToString:call.method]) {
        [_loginService handleIsLogin:call result:result];
    } else if ([@"getUser" isEqualToString:call.method]) {
        [_loginService handleGetUser:call result:result];
    } else if ([@"logout" isEqualToString:call.method]) {
        [_loginService handleLogout:result];
    } else if([@"openItemDetail" isEqualToString:call.method]){
        [_tradeHandler openItemDetail:call result:result];
    } else if([@"openUrl" isEqualToString:call.method]){
        [_tradeHandler openUrl:call result:result];
    }else if ([@"getPlatformVersion" isEqualToString:call.method]) {
        result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
    } else {
        result(FlutterMethodNotImplemented);
    }
}


- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation {
    
    return [[AlibcTradeSDK sharedInstance] application:application openURL:url sourceApplication:sourceApplication annotation:annotation];
}

// NOTE: 9.0以后使用新API接口
- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<NSString *, id> *)options {
    return [[AlibcTradeSDK sharedInstance] application:app openURL:url options:options];
}


@end

File: android/src/main/kotlin/com/hongtang/easytbk/EasytbkPlugin.kt
package com.hongtang.easytbk

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.PluginRegistry.Registrar

class EasytbkPlugin(registrar: Registrar): MethodCallHandler {
    private val loginService = LoginServiceHandler(registrar)
    private val tradeHandler = TradeHandler(registry = registrar)

  companion object {
    @JvmStatic
    fun registerWith(registrar: Registrar) {
      val channel = MethodChannel(registrar.messenger(), "easytbk")
      channel.setMethodCallHandler(EasytbkPlugin(registrar))
    }
  }

  override fun onMethodCall(call: MethodCall, result: Result) {

    when {
        "login" == call.method -> loginService.handleLogin(call, result)
        "initLogin" == call.method -> loginService.initLogin(result)
        "isLogin" == call.method -> loginService.handleIsLogin(result)
        "getUser" == call.method -> loginService.handleGetUser(result)
        "logout" == call.method -> loginService.handleLogout(result)
        "initTradeAsync" == call.method -> tradeHandler.initTradeAsync(call,result)
        "openItemDetail" == call.method -> tradeHandler.openItemDetail(call,result)
        "openUrl" == call.method -> tradeHandler.openUrl(call, result)
        "destroy" == call.method -> tradeHandler.destroy()
        "getPlatformVersion" == call.method -> result.success("Android ${android.os.Build.VERSION.RELEASE}")
        else -> result.notImplemented()
    }

    // if (call.method == "getPlatformVersion") {
    //   result.success("Android ${android.os.Build.VERSION.RELEASE}")
    // } else {
    //   result.notImplemented()
    // }
  }
}
